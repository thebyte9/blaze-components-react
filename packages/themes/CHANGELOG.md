# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.8.0-alpha.20](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.19...v0.8.0-alpha.20) (2021-07-13)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.19](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.18...v0.8.0-alpha.19) (2021-07-13)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.18](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.17...v0.8.0-alpha.18) (2021-07-13)

**Note:** Version bump only for package @blaze-react/themes







**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.17](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.16...v0.8.0-alpha.17) (2021-07-13)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.16](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.15...v0.8.0-alpha.16) (2021-07-13)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.15](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.14...v0.8.0-alpha.15) (2021-07-13)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.14](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.13...v0.8.0-alpha.14) (2021-07-13)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.13](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.12...v0.8.0-alpha.13) (2021-07-13)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.12](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.11...v0.8.0-alpha.12) (2021-07-13)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.11](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.10...v0.8.0-alpha.11) (2021-07-13)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.10](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.9...v0.8.0-alpha.10) (2021-07-13)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.9](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.8...v0.8.0-alpha.9) (2021-07-12)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.8](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.7...v0.8.0-alpha.8) (2021-07-12)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.7](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.6...v0.8.0-alpha.7) (2021-07-12)

**Note:** Version bump only for package @blaze-react/themes





# [0.8.0-alpha.6](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.3...v0.8.0-alpha.6) (2021-07-09)


### Features

* add stories for filled, affix, outlined and text buttons ([9f5cff4](https://github.com/thebyte9/blaze-components-react/commit/9f5cff45b8c6a94ebb5dd7369624bb1dbb892c10))
* add variations page ([29abb4f](https://github.com/thebyte9/blaze-components-react/commit/29abb4f048040b2e2ad029a28a6029630692819d))
