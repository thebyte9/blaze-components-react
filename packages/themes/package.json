{
  "name": "@blaze-react/themes",
  "version": "0.8.0-alpha.35",
  "description": "Blaze themes",
  "main": "./dist/main.js",
  "scripts": {
    "build": "NODE_ENV=production webpack --config webpack.config.js",
    "build-storybook": "build-storybook -c .storybook -o docs-build",
    "lint": "eslint '*/**/*.{ts,tsx}' --ignore-path ../../.eslintignore --fix",
    "test": "jest --passWithNoTests",
    "test:coverage": "jest --coverage --passWithNoTests",
    "pre-publish": "yarn build && yarn lint && yarn test",
    "bundle-analyser": "node ../../node_modules/webpack-bundle-analyzer/lib/bin/analyzer.js ./stats/stats.json"
  },
  "keywords": [],
  "author": "byte9",
  "license": "GPL-3.0",
  "dependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devDependencies": {
    "webpack": "^5.28.0",
    "webpack-cli": "^4.6.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
