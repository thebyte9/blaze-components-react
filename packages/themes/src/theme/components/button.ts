export const button = {
  variants: {
    default: {
      container: [
        'font-manrope',
        'font-medium',
        'rounded-button',
        'bg-button-primary',
        'text-button-primary',
        'px-6',
        'py-1',
        'min-w-min',
        'dark:bg-gray-800',
        'dark:text-button-primary',
        'hover:bg-button-hover',
        'focus:outline-none',
        'active:bg-button-pressed',
        'disabled:text-button-primary',
        'disabled:bg-button-disabled',
        'disabled:cursor-not-allowed',
        'disabled:border-button-disabled',
      ],
      skeleton: {
        container: ['bg-gray-400', 'rounded-button', 'w-28', 'h-8'],
      },
    },
    secondary: {
      container: [
        'font-manrope',
        'font-medium',
        'text-bold',
        'rounded-button',
        'bg-button-outlined',
        'text-button-outlined',
        'px-6',
        'py-1',
        'min-w-min',
        'dark:bg-gray-800',
        'dark:text-button-primary',
        'border-2',
        'border-button-outlined',
        'hover:bg-button-outlined-hover',
        'hover:text-button-primary',
        'focus:outline-none',
        'active:bg-button-pressed',
        'active:border-button-outlined-pressed',
        'disabled:text-button-outlined-disabled',
        'disabled:bg-transparent',
        'disabled:border-button-outlined-disabled',
        'disabled:cursor-not-allowed',
      ],
      skeleton: {
        container: ['bg-gray-400', 'rounded-button', 'w-28', 'h-8'],
      },
    },
  },
};
