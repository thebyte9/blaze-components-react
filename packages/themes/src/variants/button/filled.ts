// Note: this should be move out to the Button stories folder

export const filledVariants = {
  filled: {
    primary: `
      font-manrope
      font-medium
      rounded-button 
      bg-button-primary 
      text-button-primary 
      px-6 
      py-1 
      min-w-min 
      dark:bg-gray-800 
      dark:text-button-primary 
      hover:bg-button-hover 
      focus:outline-none
      mr-1 
      active:bg-button-pressed`,

    disabled: `
        font-manrope
        font-medium
        rounded-button 
        bg-button-disabled 
        text-button-primary 
        px-6 
        py-1 
        min-w-min 
        dark:bg-gray-800 
        dark:text-button-disabled 
        focus:outline-none
    `,

    small: `
        font-manrope
        font-medium
        rounded-button 
        bg-button-primary 
        text-button-primary 
        px-4 
        py-1 
        min-w-min 
        dark:bg-gray-800 
        dark:text-button-primary 
        hover:bg-button-hover 
        focus:outline-none 
        active:bg-button-pressed
        text-sm
    `,

    medium: `
      font-manrope
      font-medium
      rounded-button 
      bg-button-primary 
      text-button-primary 
      px-6 
      py-1 
      min-w-min 
      dark:bg-gray-800 
      dark:text-button-primary 
      text-base-primary 
      hover:bg-button-hover 
      focus:outline-none 
      active:bg-button-pressed
      text-md
    `,

    large: `
      font-manrope
      font-medium
      rounded-button-large 
      bg-button-primary 
      text-button-primary 
      px-6 
      py-1 
      min-w-min 
      dark:bg-gray-800 
      dark:text-button-primary 
      text-base-primary 
      hover:bg-button-hover 
      focus:outline-none 
      active:bg-button-pressed
      text-lg
    `,

    stretched: `
      font-manrope
      font-medium
      rounded-button 
      bg-button-primary 
      text-button-primary 
      px-6 
      py-1 
      w-full 
      dark:bg-gray-800 
      dark:text-button-primary 
      hover:bg-button-hover 
      focus:outline-none 
      active:bg-button-pressed
      text-lg
    `,

    iconLeft: `
      font-manrope
      font-medium
      rounded-button 
      bg-button-primary 
      text-button-primary 
      px-6 
      py-1 
      min-w-min 
      dark:bg-gray-800 
      dark:text-button-primary 
      hover:bg-button-hover 
      focus:outline-none 
      active:bg-button-pressed     
    `,

    iconRight: `
      font-manrope
      font-medium
      rounded-button 
      bg-button-primary 
      text-button-primary 
      px-6 
      py-1 
      min-w-min 
      dark:bg-gray-800 
      dark:text-button-primary 
      hover:bg-button-hover 
      focus:outline-none 
      active:bg-button-pressed
    `,
  },
};
