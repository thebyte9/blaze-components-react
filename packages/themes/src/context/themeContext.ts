import { createContext, useContext } from 'react';
import { ITheme } from '../types';
import { adminTheme } from '../themes/adminTheme';

export enum Themes {
  Admin = 'Admin',
  Frontend = 'Frontend',
}

type ThemeContextType = {
  theme: ITheme;
  setTheme: (theme: ITheme) => void;
};

export const ThemeContext = createContext<ThemeContextType>({
  theme: adminTheme,
  setTheme: () => console.warn('Theme provider not found'),
});

export const useTheme = (): any => useContext(ThemeContext);
