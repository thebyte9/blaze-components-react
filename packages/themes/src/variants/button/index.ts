// TODO: another iteration - reuse common class utilities
import { filledVariants } from './filled';
import { outlinedVariants } from './outlined';
import { textVariants } from './text';

export const buttonVariants = {
  ...filledVariants,
  ...outlinedVariants,
  ...textVariants,
};
