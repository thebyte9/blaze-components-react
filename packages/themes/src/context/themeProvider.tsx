import { ThemeType } from '../types';
import React from 'react';
import { ThemeContext } from './themeContext';

interface ThemeProviderProps {
  theme: ThemeType;
  showSkeleton: boolean;
  children: unknown;
  setTheme?: (theme: ThemeType) => void;
}

export function ThemeProvider({ theme, setTheme, showSkeleton, children }: ThemeProviderProps): JSX.Element {
  return (
    <ThemeContext.Provider value={{ theme: theme, setTheme: setTheme, showSkeleton: showSkeleton }}>
      {children}
    </ThemeContext.Provider>
  );
}
