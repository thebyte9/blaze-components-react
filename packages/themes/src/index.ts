import { withOpacity } from './utils/withOpacity';
import { overrideClasses } from './utils/overrideClasses';
import { adminTheme } from './themes/adminTheme';
import { frontendTheme } from './themes/frontendTheme';
import { ThemeContext, useTheme } from './context/themeContext';
import { ThemeProvider } from './context/themeProvider';
import { applyTheme, createTheme } from './utils';
import { variants } from './variants';
import { preset } from './theme/preset';
import { useVariant } from './hooks/useVariant';
import { ThemeType } from './types';

export {
  withOpacity,
  adminTheme,
  frontendTheme,
  ThemeContext,
  useTheme,
  applyTheme,
  createTheme,
  variants,
  overrideClasses,
  preset,
  useVariant,
  ThemeProvider,
  ThemeType,
};
