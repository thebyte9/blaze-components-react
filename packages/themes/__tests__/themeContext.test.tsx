import React from 'react';
import { render } from '@testing-library/react';
import { ThemeContext } from '../src/context/themeContext';
import { ThemeProvider } from '../src/context/themeProvider';
import { preset } from '@blaze-react/themes';
import { Button } from '@blaze-react/button';

describe('Theme Context', () => {
  test('it should render correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={preset} showSkeleton>
        <Button />
      </ThemeProvider>,
    );
    expect(asFragment).toMatchSnapshot();
  });

  test('it should console.warn if theme is undefined', () => {
    const setTheme = jest.fn();

    const { asFragment } = render(
      <ThemeContext.Provider
        value={{ theme: preset, setTheme: setTheme, showSkeleton: false }}
      ></ThemeContext.Provider>,
    );

    expect(asFragment).toMatchSnapshot();
  });
});
