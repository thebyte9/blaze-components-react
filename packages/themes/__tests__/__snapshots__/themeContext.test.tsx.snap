// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Context it should console.warn if theme is undefined 1`] = `[Function]`;

exports[`Theme Context it should render correctly 1`] = `[Function]`;
